snapcraft/parts/setup_assets.py

```diff
@@ -52,6 +52,7 @@ def setup_assets(
     # create wrappers for hooks in the snap/hooks directory
     _create_hook_wrappers(prime_dir)
 
+    # hooks with command chains will execute the command chain instead of the hook wrapper
```

Comment:
this comment makes no sense, the hook is still executed

Reply:
I'd appreciate some clarification on this! I can't find anything that explicitly clarifies command-chains execution for hooks.  I've been looking in the legacy codebase, snapd codebase, and in the documentation.

After testing, I think that the execution of the hook is conditional on the executables in the command-chain ending with `exec "$@"`.  This is the only way it is tested in [snapd](https://github.com/snapcore/snapd/blob/7f63a940dc584fa0ba528986daa2e83a650e0b9e/tests/main/command-chain/command-chain/chain1#L6).

Without `exec $@` at the end of the command-chain scripts, the hook itself doesn't execute.  And there are no warnings or errors.  I can remove the stub hook too, with no errors or warnings.